env/env_posix.cc

```diff
@@ -488,17 +488,17 @@ Env* Env::Default() {
   CompressionContextCache::InitSingleton();
   INIT_SYNC_POINT_SINGLETONS();
   // ~PosixEnv must be called on exit
-  static PosixEnv default_env;
+  STATIC_AVOID_DESTRUCTION(PosixEnv, default_env);
```

Comment:
AFAIK, this is our one static object that must be statically destructed. (Alternate analogous object for other platforms, I presume?) If we only have one object to statically destroy in RocksDB, and DBs are closed so not calling any callbacks, static destruction order should not matter.